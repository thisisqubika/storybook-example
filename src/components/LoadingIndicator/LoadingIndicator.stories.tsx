import type { Meta, StoryObj } from '@storybook/react';
import { ThemeWrapper } from '../../common/themes';
import { LoadingIndicator, type LoadingIndicatorProps } from '.';
import { Typography } from '@mui/material';

export default {
  component: LoadingIndicator,
} satisfies Meta<LoadingIndicatorProps>; // meta

export const Base: StoryObj<LoadingIndicatorProps> = {
  args: {
  } satisfies LoadingIndicatorProps,

  render() {
    return (
      <ThemeWrapper>
        <LoadingIndicator />
      </ThemeWrapper>
    );
  },
}; // const Base

export const Error: StoryObj<LoadingIndicatorProps> = {
  args: {
    errorMessage: (
      <Typography variant='h5' color='error'>Errare humanum est!</Typography>
    ),
  } satisfies LoadingIndicatorProps,

  render(props: LoadingIndicatorProps) {
    return (
      <ThemeWrapper>
        <LoadingIndicator {...props} />
      </ThemeWrapper>
    );
  },
}; // const Base
