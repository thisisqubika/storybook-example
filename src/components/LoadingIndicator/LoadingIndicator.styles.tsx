import { SxProps, Theme } from '@mui/material';

export function sxStyles(): Record<string, SxProps<Theme>> {
  return {
    errorIcon: {
      height: '5rem',
      width: '5rem',
    },
    
    root: {
      alignItems: 'center',
      display: 'flex',
      flexDirection: 'column',
      gap: '1rem',
      height: '50vh',
      justifyContent: 'center',
      margin: '1rem',
      padding: '1rem',
    }, // root

  };
} // sxStyles
