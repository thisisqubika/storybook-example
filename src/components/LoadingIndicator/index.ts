export {
  default as LoadingIndicator,
  type LoadingIndicatorProps,
} from './LoadingIndicator';
