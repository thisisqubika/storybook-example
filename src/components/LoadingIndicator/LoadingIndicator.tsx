import { CircularProgress, Paper, Typography } from '@mui/material';
import ErrorIcon from '@mui/icons-material/Error';
import { sxStyles } from './LoadingIndicator.styles';

export interface LoadingIndicatorProps {
  errorMessage?: React.ReactNode;
} // interface LoadingIndicatorProps

export default function LoadingIndicator({
  errorMessage,
}: LoadingIndicatorProps) {
  const styles = sxStyles();
  return (
    <Paper
      data-testid='Loading'
      sx={styles.root}
      variant='outlined'
    >
      {errorMessage ? (
        <>
          {errorMessage}
          <ErrorIcon color='error' sx={styles.errorIcon} />
        </>
      ) : (
        <>
          <Typography variant='h5'>Loading, please wait...</Typography>
          <CircularProgress size='5rem' />
        </>
      )}
    </Paper>
  );
} // function Loading
