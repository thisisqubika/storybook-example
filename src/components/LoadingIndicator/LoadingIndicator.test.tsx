import { describe, expect, test } from 'vitest';
import { composeStories } from "@storybook/react";
import { render } from '@testing-library/react';
import * as stories from './LoadingIndicator.stories';
import { type LoadingIndicatorProps } from './LoadingIndicator';

const {
  Base: LoadingIndicatorBaseStory,
  Error: LoadingIndicatorErrorStory,
} = composeStories(stories);

describe('LoadingIndicator', () => {
  test('stories match snapshot', () => {
    for (const Story of [
      LoadingIndicatorBaseStory,
      LoadingIndicatorErrorStory,
    ]) {
      const props = Story.args as LoadingIndicatorProps;
      const { container } = render(
        <Story {...props} />
      );
      expect(container).toMatchSnapshot();
    }
  }); // test

}); // describe 'LoadingIndicator'
