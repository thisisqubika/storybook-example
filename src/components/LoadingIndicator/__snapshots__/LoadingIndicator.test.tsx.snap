// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingIndicator > stories match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-1qbral5-MuiPaper-root"
    data-testid="Loading"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1oepjcl-MuiTypography-root"
    >
      Loading, please wait...
    </h5>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18li798-MuiCircularProgress-root"
      role="progressbar"
      style="width: 5rem; height: 5rem;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`LoadingIndicator > stories match snapshot 2`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-1qbral5-MuiPaper-root"
    data-testid="Loading"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-ojno0f-MuiTypography-root"
    >
      Errare humanum est!
    </h5>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1y7yv10-MuiSvgIcon-root"
      data-testid="ErrorIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-2h2zm0-4h-2V7h2z"
      />
    </svg>
  </div>
</div>
`;
